File: codegen/results/v002/m3-max-48gb/lmstudio/qwen3.8-27b-mlx-4bit_think/20260830_233604/go/005_unit_test_writer/solution.go
package main

import "fmt"

func calculateDiscount(price float64, quantity int) float64 {
	if quantity < 10 {
		return price * float64(quantity)
	} else if quantity < 50 {
		return price * float64(quantity) * 0.9
	}
	return price * float64(quantity) * 0.75
}

func almostEqual(a, b float64) bool {
	d := a - b
	if d < 0 {
		d = -d
	}
	return d < 1e-9
}

type testCase struct {
	description string
	price       float64
	quantity    int
	expected    float64
}

func main() {
	tests := []testCase{
		{"quantity below 10 (no discount)", 100.0, 9, 900.0},
		{"quantity exactly 10 (10% discount)", 100.0, 10, 900.0},
		{"quantity 49 (10% discount)", 100.0, 49, 4410.0},
		{"quantity exactly 50 (25% discount)", 100.0, 50, 3750.0},
	}

	for _, tc := range tests {
		got := calculateDiscount(tc.price, tc.quantity)
		if almostEqual(got, tc.expected) {
			fmt.Printf("PASS: %s\n", tc.description)
		} else {
			fmt.Printf("FAIL: %s (expected: %g, got: %g)\n", tc.description, tc.expected, got)
		}
	}
}

File: codegen/results/v002/m3-max-48gb/lmstudio/qwen3.8-27b-mlx-4bit_think/20260830_233604/typescript/005_unit_test_writer/solution.ts
function calculateDiscount(price: number, quantity: number): number {
    if (quantity < 10) return price * quantity;
    else if (quantity < 50) return price * quantity * 0.9;
    else return price * quantity * 0.75;
}

function runTest(
    description: string,
    price: number,
    quantity: number,
    expected: number
): void {
    const got = calculateDiscount(price, quantity);
    const tolerance = 1e-9;

    if (Math.abs(got - expected) < tolerance) {
        console.log(`PASS: ${description}`);
    } else {
        console.log(`FAIL: ${description} (expected: ${expected}, got: ${got})`);
    }
}

// Boundary: just below 10 (no discount)
runTest("price=10, quantity=9 (no discount)", 10, 9, 10 * 9);

// Boundary: exactly 10 (10% discount kicks in)
runTest("price=10, quantity=10 (10% discount)", 10, 10, 10 * 10 * 0.9);

// Boundary: just below 50 (still 10% discount)
runTest("price=10, quantity=49 (10% discount)", 10, 49, 10 * 49 * 0.9);

// Boundary: exactly 50 (25% discount kicks in)
runTest("price=10, quantity=50 (25% discount)", 10, 50, 10 * 50 * 0.75);

// Extra: mid-range with different price
runTest("price=25, quantity=20 (10% discount)", 25, 20, 25 * 20 * 0.9);

// Extra: large quantity with different price
runTest("price=5, quantity=100 (25% discount)", 5, 100, 5 * 100 * 0.75);